File: data/EXP01/python/2241/2241.output_2.py
# @lc app=leetcode id=2241 lang=python3
class ATM:

    def __init__(self):
        self.banknotesCount = [0, 0, 0, 0, 0]

    def deposit(self, banknotesCount):
        for i in range(5):
            self.banknotesCount[i] += banknotesCount[i]

    def withdraw(self, amount):
        ans = [0, 0, 0, 0, 0]
        values = [500, 200, 100, 50, 20]

        for i in range(5):
            while self.banknotesCount[i] > 0 and amount >= values[i]:
                amount -= values[i]
                self.banknotesCount[i] -= 1
                ans[i] += 1

        if amount != 0:
            return [-1]

        return ans

File: data/EXP01/golang_no_system_var/2241/2241.output_2.go
type ATM struct {
    banknotesCount []int
}

func Constructor() ATM {
    return ATM{banknotesCount: make([]int, 5)}
}

func (a *ATM) Deposit(banknotesCount []int) {
    for i := 0; i < 5; i++ {
        a.banknotesCount[i] += banknotesCount[i]
    }
}

func (a *ATM) Withdraw(amount int) []int {
    withdrawnCount := make([]int, 5)
    denominations := []int{500, 200, 100, 50, 20}

    for i := 0; i < 5; i++ {
        withdrawnCount[i] = min(amount/denominations[i], a.banknotesCount[i])
        amount -= withdrawnCount[i] * denominations[i]
    }

    if amount > 0 {
        return []int{-1}
    }

    for i := 0; i < 5; i++ {
        a.banknotesCount[i] -= withdrawnCount[i]
    }

    return withdrawnCount
}

func min(a, b int) int {
    if a < b {
        return a
    }
    return b
}
